src/common/benches/bench_row.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright 2023 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use risingwave_common::catalog::ColumnId;
+use risingwave_common::error::Result;
+use risingwave_common::row::{OwnedRow, Row, RowDeserializer};
+use risingwave_common::types::{DataType, Datum, ScalarImpl};
+use risingwave_common::util::ordered::OrderedRowSerde;
+use risingwave_common::util::sort_util::OrderType;
+use risingwave_common::util::value_encoding::row_encoding;
+
+struct Case {
+    name: String,
+    schema: Vec<DataType>,
+    column_ids: Vec<ColumnId>,
+    rows: Vec<OwnedRow>,
+    needed_schema: Vec<DataType>,
+    needed_ids: Vec<ColumnId>,
+}
+
+impl Case {
+    pub fn new(
+        name: &str,
+        schema: Vec<DataType>,
+        column_ids: Vec<ColumnId>,
+        rows: Vec<OwnedRow>,
+        needed_schema: Option<Vec<DataType>>,
+        needed_ids: Option<Vec<ColumnId>>,
+    ) -> Self {
+        Self {
+            name: name.to_string(),
+            schema: schema.clone(),
+            column_ids: column_ids.clone(),
+            rows,
+            needed_ids: needed_ids.unwrap_or(column_ids.clone()),
+            needed_schema: needed_schema.unwrap_or(schema.clone()),
+        }
+    }
+}
+
+fn memcmp_encode(c: &Case) -> Vec<Vec<u8>> {
+    let serde = OrderedRowSerde::new(
+        c.schema.clone(),
+        vec![OrderType::Descending; c.schema.len()],
+    );
+    let mut array = vec![];
+    for row in &c.rows {
+        let mut row_bytes = vec![];
+        black_box(serde.serialize(row, &mut row_bytes));
+        array.push(row_bytes);
+    }
+    array
+}
+
+fn basic_encode(c: &Case) -> Vec<Vec<u8>> {
+    let mut array = vec![];
+    for row in &c.rows {
+        let row_encoding = black_box(row.value_serialize());
+        array.push(row_encoding);
+    }
+    array
+}
+
+fn column_aware_encode(c: &Case) -> Vec<Vec<u8>> {
+    let seralizer = row_encoding::Serializer::new(&c.column_ids);
+    let mut array = vec![];
+    for row in &c.rows {
```

Comment:
We should have more rows (>= 1000) here for benchmark, or constructing the serializer cannot be amortized.